staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -332,15 +332,22 @@ Loop:
 
 // parseFilter scans filter inside array selection
 func (p *Parser) parseFilter(cur *ListNode) error {
+	closure := 0
 	p.pos += len("[?(")
 	p.consumeText()
 Loop:
 	for {
 		switch p.next() {
 		case eof, '\n':
 			return fmt.Errorf("unterminated filter")
+		case '(':
+			closure = closure + 1
```

Comment:
This only works if the string you are trying to match against contains paired parentheses, with open parens before close parens.
- "(" won't break the loop early enough
- ")" will still break too early
- ")(" will still break too early

You need to track when you're inside the quotes surrounding the string to match against.